### PR TITLE
chore(docker): Add prod build stage to the Dockerfile

### PR DESCRIPTION
**Pull request by ERosendo**
This PR adds a new stage to the docker build called `web-prod`. This new stage uses **gunicorn** to serve the project. the `docker-compose.yml` file is also updated to use the stage called `web-dev` to set up a development machine.

This PR resolves https://github.com/freelawproject/bigcases2/issues/4

### CODE REVIEW

File: docker/django/Dockerfile [hunk 1]
```diff
@@ -50,7 +50,60 @@ RUN poetry install
...
+    --group www-data \
+    # Set high number of workers. Docs recommend 2-4× core count`
+    --workers ${NUM_WORKERS:-48} \
+    # Allow longer queries to solr.
+    --limit-request-line 6000 \
+    # Reset each worker once in a while
+    --max-requests 10000 \
+    --max-requests-jitter 100 \
+    --timeout 180 \
+    --bind 0.0.0.0:8000
```
Comment (mlissner) [inline docker/django/Dockerfile:109]:
```suggestion
    --bind 0.0.0.0:8888
```

File: docker/django/Dockerfile [hunk 2]
```diff
@@ -50,7 +50,60 @@ RUN poetry install
...
+
+CMD python /opt/bigcases2/manage.py migrate  && \
+    gunicorn bc2_wsgi:application \
+    --chdir /opt/bigcases2/docker/django/wsgi-configs/ \
+    --user www-data \
+    --group www-data \
+    # Set high number of workers. Docs recommend 2-4× core count`
+    --workers ${NUM_WORKERS:-48} \
+    # Allow longer queries to solr.
+    --limit-request-line 6000 \
```
Comment (mlissner) [inline docker/django/Dockerfile:104]: I don't think we need the Solr stuff, and in k8s you use smaller machines, so:

```suggestion
    # Set high number of workers. Docs recommend 2-4× core count`
    --workers ${NUM_WORKERS:-16} \
```

**Review by mlissner — COMMENTED** [review]
Couple little things, but I assume 95% of this is just copied over from CL? If so, I guess it's fine, though complex. 

Once tweaked, feel free to merge.

**Comment by ERosendo** [comment]
Yes. I use the dockerfile from CL as a template.

**Pull request merged**